### PR TITLE
"Call `atexit.register(_flush_queue)` in `tensorflow.autolog`

### LINKED ISSUE #3400: [BUG] - `mlruns` directory automatically created without doing anything
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS 10.15.4 and Linux
- **MLflow installed from (source or binary)**: conda-forge
- **MLflow version (run ``mlflow --version``)**: 1.11
- **Python version**: tested on 3.8 and 3.7
- **npm version, if running the dev UI**:
- **Exact command to reproduce**: `import mlflow`

### Describe the problem
With version `1.11`, simply importing `mlflow` results in an `mlruns` directory being created.

### Code to reproduce issue
Here is an anaconda env:
```yaml
name: ml
channels:
  - conda-forge
dependencies:
  - mlflow=1.11
```

Activating (`conda activate ml`) and then importing `mlflow` causes the `mlruns` directory from being created
```python
import mlflow
print(f"mlflow version {mlflow.__version__}")
```
Prints `mlflow version 1.11.0`.

This is BAD because it means that any project that uses `mlflow` in its backend creates these `mlruns` directories all over the place. In my specific example, I have `mlflow` as a dependency and these directories are created within my test suite.

Note that even running `mlflow --version` in the terminal results in the `mlruns` directory being created.

My proposed solution is to not create this directory by default.

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Fixes #3400 

When importing the tensorflow flavor, `mlruns` folder is created by `_flush_queue` registered by `atexit.register`. 

https://github.com/mlflow/mlflow/blob/e34dbf236f7c19e9774365d03c87bc1b746d932f/mlflow/tensorflow.py#L577-L591

We currently import the tensorflow flavor in `mlflow.__init__.py`. This means executing `import mlflow` creates `mlruns` folder. To fix this issue, call `atexit.register(_flush_queue)` in the `tensorflow.autolog` function.

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -197,6 +197,12 @@ jobs:
         run: |
           python -c "import mlflow"
 
+          # Just importing mlflow should not create `mlruns` directory
+          # See: https://github.com/mlflow/mlflow/issues/3400
+          if [ -d "./mlruns" ]; then
+            exit 1
+          fi
```
Comment (harupy) [inline .github/workflows/master.yml:204]: @tmcclintock If `python -c "import mlflow"` still created the `mlruns` directory, this check would fail.

Comment (tmcclintock) [reply .github/workflows/master.yml:204]: As I wrote in the issue: I was accidentally had `master` installed when I tried the test. Your fix works. Thanks again.

**Comment by tmcclintock** [comment]
Any chance we can get this merged?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Comment by tmcclintock** [comment]
@harupy @dbczumar any idea when this fix will end up in a tagged version? I'd like to update some other project environments asap with this fix. Thanks again!

**Pull request merged**